browser/brave_stats/first_run_util.cc

```diff
@@ -12,22 +12,34 @@
 
 namespace brave_stats {
 
-base::Time GetFirstRunTime(PrefService* local_state) {
 #if BUILDFLAG(IS_ANDROID)
+namespace {
+
+bool g_is_first_run = false;
+
+base::Time InitFirstRunTime(PrefService* local_state) {
+  base::Time now = base::Time::Now();
+  local_state->SetTime(kReferralAndroidFirstRunTimestamp, now);
+  g_is_first_run = true;
+  return now;
+}
+
+}  // namespace
+#endif
+
+base::Time GetFirstRunTime([[maybe_unused]] PrefService* local_state) {
```

Comment:
nit: we could return std::optional<base::Time> and then call `InitFirstRunTime` outside of the getter.